src/components/TheTableOfContents.vue

```diff
@@ -0,0 +1,253 @@
+<template>
+  <AppFlex
+    direction="col"
+    gap="none"
+    hAlign="stretch"
+    class="toc"
+    :style="{ top: nudge + 'px' }"
+    :data-expanded="expanded"
+    role="doc-toc"
+    aria-label="Page table of contents"
+    @click.stop
+  >
+    <!-- toggle button -->
+    <AppFlex gap="none" hAlign="left" class="title">
+      <button
+        class="button"
+        @click="expanded = !expanded"
+        :aria-expanded="expanded"
+        v-tippy="
+          expanded ? 'Close table of contents' : 'Expand table of contents'
+        "
+      >
+        <AppIcon :icon="expanded ? 'times' : 'bars'" />
+      </button>
+      <span v-if="expanded" class="title-text">Table of Contents</span>
+    </AppFlex>
+
+    <template v-if="expanded">
+      <div class="spacer"></div>
+
+      <!-- entries -->
+      <AppLink
+        v-for="(entry, index) in entries"
+        :key="index"
+        :to="'#' + entry.id"
```

Comment:
IMHO, the history stack gets a little long when navigating around the sections pushes items onto the history, as it does by default with `:to`. Instead, it might be nicer to replace the current history entry (e.g., https://router.vuejs.org/guide/essentials/navigation.html#replace-current-location), then it's IMO easier to get back to the original list.

Reply:
Good catch. For some reason I thought `vue-router`'s link component did that automatically when the `to` is just a hash.

Reply:
Check that my latest commit works. Added this behavior by default to the app link component, as we'll always want that for hash links.

Also added the use of the app link component in a few places where I (incorrectly) wasn't before.

Reply:
Nice, it seems like it's working. I noticed while I was testing it, though, that opening a page with a fragment doesn't move to that section of the page -- I hope the change I suggested didn't break that! For example, I'd expect https://deploy-preview-92--monarch-ui-new.netlify.app/disease/MONDO:0009120#breadcrumbs to take me to the breadcrumbs section, but it just stays at the top.